### PR TITLE
Remove docker-compose from devstack

### PR DESCRIPTION
**Pull request by rgraber**
Replace docker-compose with docker compose in all instances. Docker-compose is no longer supported. For macs, this requires installing the docker compose plugin.
Also adds provisioning tests for analyticsapi and insights to the PR checks. Coursegraph needs its neo4j image updated before tests will pass so I left it out.

----

I've completed each of the following or determined they are not applicable:

- [ ] Made a plan to communicate any major developer interface changes (or N/A)

### CODE REVIEW

File: .github/workflows/cli-tests.yml [hunk 1]
```diff
@@ -57,6 +57,9 @@ jobs:
           brew install lima docker
           limactl start --name=default template://docker
           echo "DOCKER_HOST=unix:///Users/runner/.lima/default/sock/docker.sock" >> $GITHUB_ENV
+          # install the docker compose plugin
+          mkdir -p ~/.docker/cli-plugins
```
Comment (rgraber) [inline .github/workflows/cli-tests.yml:60]: This is where the docker compose plugin is made discoverable

File: .github/workflows/provisioning-tests.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ jobs:
         os:
           - ubuntu-20.04 # Ubuntu 20.04 "Focal Fossa"
         python-version: [ '3.8' ]
-        services: [ discovery+lms+forum ,registrar+lms, ecommerce+lms, edx_notes_api+lms, credentials+lms, xqueue]
+        services: [ discovery+lms+forum ,registrar+lms, ecommerce+lms, edx_notes_api+lms, credentials+lms, xqueue, analyticsapi+insights+lms]
```
Comment (rgraber) [inline .github/workflows/provisioning-tests.yml:31]: Add analytics and insights to provisioning tests

File: requirements/base.in [hunk 3]
```diff
@@ -1,4 +1,4 @@
 -c constraints.txt
 
-docker-compose                      # For launching the devstack Docker containers
```
Comment (rgraber) [inline requirements/base.in:3]: Remove the PyPI docker-compose, which pinned PyYAML to a version that didn't work on Apple Silicon

File: .github/workflows/cli-tests.yml [hunk 4]
```diff
@@ -48,7 +48,7 @@ jobs:
           sudo apt update
           sudo apt install docker-ce containerd.io
           docker version
-          docker-compose --version
+          docker compose --version
```
Comment (johnnagro) [inline .github/workflows/cli-tests.yml:51]: [inform] this spaced version also works with podman
```
➜  docker compose --version
podman version 4.6.2
```

**Review by johnnagro — APPROVED** [review]

**Review by dianakhuang — APPROVED** [review]
I think this looks good. Feel free to wait for Tim to test on Linux and verify that we don't need to do any soft linking shenanigans there.

**Comment by timmc-edx** [comment]
`docs/getting_started.rst` has a "Prerequisites" section that describes what you need to install for Linux and Mac. I think we'll want to update that as well.

For Linux, developers on Ubuntu (and Debian) should ensure they've uninstalled `docker.io` and `docker-compose` from the main Ubuntu repositories (this is why I was inadvertently using an old version) and instead install `docker-ce` and `docker-compose-plugin` from the official Docker package repository: https://docs.docker.com/engine/install/ubuntu/ -- not sure what the minimum version would be, although maybe we don't need to include that.

**Comment by timmc-edx** [comment]
Optionally, we may also want to give a bit of advice somewhere in the docs on how to work with older versions of devstack that still use `docker-compose`. What works for me on Linux, and will probably also work on Mac, is to make this tiny shell script and put it on the PATH under the name `docker-compose`:

```
#!/bin/bash

docker compose "$@"
```

**Comment by rgraber** [comment]
> Optionally, we may also want to give a bit of advice somewhere in the docs on how to work with older versions of devstack that still use `docker-compose`. What works for me on Linux, and will probably also work on Mac, is to make this tiny shell script and put it on the PATH under the name `docker-compose`:
> 
> ```
> #!/bin/bash
> 
> docker compose "$@"
> ```

Do we want to give advice for older devstack or just add to the solve-by-updating section of troubleshooting?

**Comment by timmc-edx** [comment]
I was actually thinking of mentioning it in the "Developing on Open edX named release branches" file, but it's probably worth mentioning on the troubleshooting page too, yeah.

**Comment by rgraber** [comment]
> I was actually thinking of mentioning it in the "Developing on Open edX named release branches" file, but it's probably worth mentioning on the troubleshooting page too, yeah.

There's already a recommendation in there for keeping devstack on master. I worry adding stuff about workarounds for older devstacks will make things more confusing rather than less. I'm also not sure what exact bug you'd get if you didn't update devstack (to put in the troubleshooting guide).

**Comment by timmc-edx** [comment]
The docs I'm looking at say to [check out the named-release branch of devstack](https://edx.readthedocs.io/projects/open-edx-devstack/en/latest/developing_on_named_release_branches.html) rather than staying on master. That means that if someone wants to work on Palm, they check out `open-release/palm.master`, and now the Makefile is referencing `docker-compose`. So if someone wants to work on both Palm and Quince, they'll need both `docker-compose` and `docker compose` to work.

**Comment by rgraber** [comment]
> The docs I'm looking at say to [check out the named-release branch of devstack](https://edx.readthedocs.io/projects/open-edx-devstack/en/latest/developing_on_named_release_branches.html) rather than staying on master. That means that if someone wants to work on Palm, they check out `open-release/palm.master`, and now the Makefile is referencing `docker-compose`. So if someone wants to work on both Palm and Quince, they'll need both `docker-compose` and `docker compose` to work.

Gotcha. Will update

**Review by timmc-edx — APPROVED** [review]

**Pull request merged**